search/search-client-connect/src/test/java/io/camunda/search/connect/plugin/PluginRepositoryTest.java

```diff
@@ -0,0 +1,179 @@
+  }
+
+  @Test
+  void shouldLoadBasePathPlugin() throws ExternalJarLoadException {
+    final var config =
+        new PluginConfiguration("PLG1", DatabaseCustomHeaderSupplierTestImpl.class.getName(), null);
+    final var jarRepository = Mockito.mock(ExternalJarRepository.class);
+    final var basePath = Mockito.mock(Path.class);
+    final var classLoader = Mockito.mock(ExternalJarClassLoader.class);
+    when(basePath.resolve(config.jarPath())).thenReturn(basePath);
+    when(jarRepository.load(basePath)).thenReturn(classLoader);
+    final var repo = new PluginRepository(new LinkedHashMap<>(), jarRepository, basePath);
+
+    repo.load(List.of(config));
+
+    final var plugins = repo.getPlugins();
+    assertThat(plugins).hasSize(1);
+  }
+
+  @Test
+  void shouldRaiseExceptionWhenClassNotFound() throws ExternalJarLoadException {
+    final var config = new PluginConfiguration("PLG1", "io.camunda.UnknownClass", null);
+    final var jarRepository = Mockito.mock(ExternalJarRepository.class);
+    final var basePath = Mockito.mock(Path.class);
+    when(basePath.resolve(config.jarPath())).thenReturn(basePath);
+    when(jarRepository.load(basePath)).thenThrow(ExternalJarLoadException.class);
+
+    final var repo = new PluginRepository(new LinkedHashMap<>(), jarRepository, basePath);
+
+    assertThatCode(() -> repo.load(List.of(config))).isInstanceOf(PluginLoadException.class);
+  }
+
+  @Test
+  void shouldRaiseExceptionWhenLoadingWrongClass() throws ExternalJarLoadException {
+    final var config = new PluginConfiguration("PLG1", "io.camunda.UnknownClass", null);
+    final var jarRepository = Mockito.mock(ExternalJarRepository.class);
+    final var basePath = Mockito.mock(Path.class);
+    final var classLoader = Mockito.mock(ExternalJarClassLoader.class);
+    when(basePath.resolve(config.jarPath())).thenReturn(basePath);
+    when(jarRepository.load(basePath)).thenReturn(classLoader);
+
+    final var repo = new PluginRepository(new LinkedHashMap<>(), jarRepository, basePath);
+
+    assertThatCode(() -> repo.load(List.of(config))).isInstanceOf(PluginLoadException.class);
+  }
+
+  @Test
+  void shouldCloseJarRepositoryWhenClosedExplicitly() throws Exception {
+    final var jarRepository = Mockito.mock(ExternalJarRepository.class);
+    final var basePath = Mockito.mock(Path.class);
+    final var repo = new PluginRepository(new LinkedHashMap<>(), jarRepository, basePath);
+
+    repo.close();
+
+    verify(jarRepository).close();
+  }
+
+  @Test
+  void shouldCloseRepositoryWhenClosedViaTryWithResources() throws Exception {
+    final var jarRepository = Mockito.mock(ExternalJarRepository.class);
+    final var basePath = Mockito.mock(Path.class);
+    try (final var repo = new PluginRepository(new LinkedHashMap<>(), jarRepository, basePath)) {
+      repo.getPlugins(); // no-op
+    }
+
+    verify(jarRepository).close();
+  }
+
+  public static final class DatabaseCustomHeaderSupplierTestImpl
```

Comment:
:wrench: Prefix with Test instead of suffix, it's a convention we tend to use to indicate clearly that this is a test impl and not production (helps, for example, when looking up indexed classes). Additionally, the `Test` suffix tends to be reserved for test classes.

Reply:
Done + moved in separate class